lib/facter/ipaddress.rb

```diff
@@ -28,15 +30,10 @@
   confine :kernel => :linux
   setcode do
     ip = nil
-    output = Facter::Util::IP.exec_ifconfig(["2>/dev/null"])
-    if output
-      regexp = /inet (?:addr:)?([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)/
-      output.split("\n").each do |line|
-        match = regexp.match(line)
-        if match and not /^127\./.match(match[1])
-          ip = match[1]
-          break
-        end
+    File.read("/proc/net/route").split("\n").each do |line|
+      parts = line.split()
+      if parts[1] == "00000000"
+        ip ||= Facter.value("ipaddress_#{parts[0]}".intern)
```

Comment:
meh. The route table shouldn't be too big, and `|||=` ensures it'll take the first one. I'd kind rather not complicate the code with extra conditionals